[Doc] Add links to external libraries: Taichi GLSL & Taichi Elements (#1250)
taichi-dev/taichi: docs/external_libraries.rst

Extension libraries
===================

The Taichi programming language offers a minimal and generic built-in standard library. Extra domain-specific functionalities are provided via **extension libraries**:

Taichi GLSL
-----------

`Taichi GLSL <https://github.com/taichi-dev/taichi_glsl>`_ is an extension
library of Taichi, aiming at providing useful helper functions including:

1. Handy scalar functions like ``clamp``, ``smoothstep``, ``mix``, ``round``.
2. GLSL-alike vector functions like ``normalize``, ``distance``, ``reflect``.

> 3. Well-behaved random generators including ``randUnit2D``, ``randNDRange``.

Reviewer: Would "random number generators" be better here? :D

Author: The rand unit in Taichi GLSL also supports random vectors, not sure if number will imply scalars..

External libraries
==================

The Taichi Programming Language provides a set of built-in standard library.
The Taichi built-in library is aimed to be simple for maintainability, and
therefore generic and less domain specific.

> However users may find that the built-in library along may not enough,

Reviewer 2: ```suggestion
However, users may find it not adequate enough for their use cases to use the built-in library alone
```

> so here we provide some external libraries that hopefully could extend your

Reviewer 2: ```suggestion
so here we provide some external libraries that hopefully could increase your
```

> The Taichi Programming Language provides a set of built-in standard library.
> The Taichi built-in library is aimed to be simple for maintainability, and
> therefore generic and less domain specific.
> 
> However users may find that the built-in library along may not enough,
> so here we provide some external libraries that hopefully could extend your
> productivity in Taichi.

Reviewer 3: ```suggestion
The Taichi programming language offers a minimal and generic built-in standard library. Extra domain-specific functionalities are provided via **extension libraries**:
```

Taichi GLSL
-----------

> `Taichi GLSL <https://github.com/taichi-dev/taichi_glsl>`_ is an external
> library of Taichi, aimed to provide a set of useful helper functions including
> but not limited to:

Reviewer 3: ```suggestion
`Taichi GLSL <https://github.com/taichi-dev/taichi_glsl>`_ is an extension
library of Taichi, aiming at providing useful helper functions including:
```

1. Handy scalar functions like ``clamp``, ``smoothstep``, ``mix``, ``round``.
2. GLSL-alike vector functions like ``normalize``, ``distance``, ``reflect``.
3. Well-behaved random generators including ``randUnit2D``, ``randNDRange``.

> 4. Possible Taichi BUG hotfixes that are not yet released in it's cycle.

Reviewer 3: Maybe this is not really the job that an extension library should do?

Author: But that's what Taichi GLSL **have to do** before v0.6.11. Sometimes Taichi BUGs can limit Taichi GLSL's ability, e.g. docstring this time.

5. Handy vector and matrix initializer: ``vec`` and ``mat``.

Click here for `Taichi GLSL Documentation <https://taichi-glsl.readthedocs.io>`_.

Taichi Elements
---------------

`Taichi Elements <https://github.com/taichi-dev/taichi_elements>`_ is a
High-Performance Multi-Material Continuum Physics Engine (work in progress).

The solver is being developed using Taichi, therefore it is cross-platform and
supports multithreaded CPUs and massively parallel GPUs.

The short-term plan is:

- To build a reusable multimaterial (water/elastic/snow/sand/mud) simulator
- To integrate the simulator into Blender

> Click here for `Taichi Elements Documentation <https://taichi-elements.readthedocs.io>`_.

Author: Wasn't it a helpful continum simulation infrastructure? Sorry, not sure about the work there...
